internal/repositories/queries.go

```diff
@@ -0,0 +1,262 @@
+	StringValueCol = "ValueString"
+
+	// IntervalGroup is the column alias for the interval group.
+	IntervalGroup = "group_timestamp"
+)
+
+var sourceTranslations = map[string]string{
+	"macron":   "dimo/integration/2ULfuC8U9dOqRshZBAi0lMM1Rrx",
+	"tesla":    "dimo/integration/26A5Dk3vvvQutjSyF0Jka2DP5lg",
+	"autopi":   "dimo/integration/27qftVRWQYpVDcO5DltO5Ojbjxk",
+	"smartcar": "dimo/integration/22N2xaPOq2WW2gAHBHd0Ikn4Zob",
+}
+
+var dialect = drivers.Dialect{
+	// Clickhouse does not like quotes around identifiers, so we set them to spaces since LQ and RQ are not optional.
+	LQ: ' ',
+	RQ: ' ',
+
+	UseIndexPlaceholders:    false,
+	UseLastInsertID:         false,
+	UseSchema:               false,
+	UseDefaultKeyword:       false,
+	UseAutoColumns:          false,
+	UseTopClause:            false,
+	UseOutputClause:         false,
+	UseCaseWhenExistsClause: false,
+}
+
+// newQuery initializes a new Query using the passed in QueryMods.
+func newQuery(mods ...qm.QueryMod) (string, []any) {
+	q := &queries.Query{}
+	queries.SetDialect(q, &dialect)
+	qm.Apply(q, mods...)
+	return queries.BuildQuery(q)
+}
+
+// selectValue adds a SELECT clause to the query to select a specific value column.
+// Example: SELECT valueCol as value.
+func selectValue(valueCol string) qm.QueryMod {
+	return qm.Select(valueCol + " as value")
+}
+
+// selectTimestamp adds a SELECT clause to the query to select the Timestamp column.
+// Example: 'SELECT Timestamp'.
+func selectTimestamp() qm.QueryMod {
+	return qm.Select("Timestamp")
+}
+
+// fromSignal adds a FROM clause to the query to select the signal table.
+// Example: 'FROM signal'.
+func fromSignal() qm.QueryMod {
+	return qm.From("signal")
+}
+
+// withTokenID adds a WHERE clause to the query to filter by TokenID.
+// Example: 'WHERE TokenID = ?'.
+func withTokenID(tokenID uint32) qm.QueryMod {
+	return qm.Where("TokenID = ?", tokenID)
+}
+
+// withName adds a WHERE clause to the query to filter by Name.
+// Example: 'WHERE Name = ?'.
+func withName(name string) qm.QueryMod {
+	return qm.Where("Name = ?", name)
+}
+
+// withFromTS adds a WHERE clause to the query to filter by Timestamp greater than fromTS.
+// Example: 'WHERE Timestamp > ?'.
+func withFromTS(fromTS time.Time) qm.QueryMod {
+	return qm.Where("Timestamp > ?", fromTS)
+}
+
+// withToTS adds a WHERE clause to the query to filter by Timestamp less than toTS.
+// Example: 'WHERE Timestamp < ?'.
+func withToTS(toTS time.Time) qm.QueryMod {
+	return qm.Where("Timestamp < ?", toTS)
+}
+
+// withSource adds a WHERE clause to the query to filter by Source.
+// Example: 'WHERE Source = ?'.
+func withSource(source string) qm.QueryMod {
+	if translateSource, ok := sourceTranslations[source]; ok {
+		return qm.Where("Source = ?", translateSource)
+	}
+	return qm.Where("Source = ?", source)
+}
+
+// orderByTS adds an ORDER BY clause to the query to order by Timestamp in descending order.
+// Example: 'ORDER BY Timestamp DESC'.
+func orderByTS() qm.QueryMod {
```

Comment:
Could we make the direction clear in the function name? This one is DESC, the next one is ASC, I'm scared.